lib/src/render/table/table_context_menu_item.dart

```diff
@@ -0,0 +1,217 @@
+    },
+  ),
+  ContextMenuItem(
+    name: 'Duplicate Row',
+    isApplicable: _isSelectionInTable,
+    onPressed: (editorState) {
+      var node = _getTableCellNode(editorState);
+      final transaction = editorState.transaction;
+      duplicateRow(
+        node.parent!,
+        node.attributes['rowPosition'],
+        transaction,
+      );
+      editorState.apply(transaction);
+    },
+  ),
+  ContextMenuItem(
+    name: 'Column Background Color',
+    isApplicable: _isSelectionInTable,
+    onPressed: (editorState) {
+      var node = _getTableCellNode(editorState);
+      _showColorMenu(
+        node,
+        editorState,
+        node.attributes['colPosition'],
+        setColBgColor,
+      );
+    },
+  ),
+  ContextMenuItem(
+    name: 'Row Background Color',
+    isApplicable: _isSelectionInTable,
+    onPressed: (editorState) {
+      var node = _getTableCellNode(editorState);
+      _showColorMenu(
+        node,
+        editorState,
+        node.attributes['rowPosition'],
+        setRowBgColor,
+      );
+    },
+  ),
+];
+
+bool _isSelectionInTable(EditorState editorState) {
+  var selection = editorState.service.selectionService.currentSelection.value;
+  if (selection == null || !selection.isSingle) {
+    return false;
+  }
+
+  var node = editorState.service.selectionService.currentSelectedNodes.first;
+
+  return node.id == kTableCellType || node.parent?.type == kTableCellType;
+}
+
+Node _getTableCellNode(EditorState editorState) {
+  var node = editorState.service.selectionService.currentSelectedNodes.first;
+  return node.id == kTableCellType ? node : node.parent!;
+}
+
+OverlayEntry? _colorMenuOverlay;
+EditorState? _editorState;
+
+void _showColorMenu(
+  Node node,
+  EditorState editorState,
+  int rowcol,
+  void Function(Node, int, Transaction, String?) action,
+) {
+  late Rect matchRect = node.rect;
+
+  _dismissColorMenu();
+  _editorState = editorState;
+
+  final style = editorState.editorStyle;
+  _colorMenuOverlay = OverlayEntry(
+    builder: (context) {
+      return Positioned(
+        top: matchRect.bottom - 15,
+        left: matchRect.left + 15,
+        child: Material(
+          color: Colors.transparent,
+          child: ColorPicker(
+            pickerBackgroundColor:
+                style.selectionMenuBackgroundColor ?? Colors.white,
+            pickerItemHoverColor: style.selectionMenuItemSelectedColor ??
+                Colors.blue.withOpacity(0.3),
+            pickerItemTextColor:
+                style.selectionMenuItemTextColor ?? Colors.black,
+            colorOptionLists: [
+              ColorOptionList(
+                header: 'background color',
+                selectedColorHex: node.attributes['backgroundColor'],
+                colorOptions: generateBackgroundColorOptions(editorState),
+                onSubmittedAction: (color) {
+                  final transaction = editorState.transaction;
+                  var c = color == node.attributes['backgroundColor']
```

Comment:
```suggestion
                  final c = color == node.attributes['backgroundColor']
```
Not sure what `c` should stand for here? maybe use eg. `is` or `has` like `final hasBgColor` for readability